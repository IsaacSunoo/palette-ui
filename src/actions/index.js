export const setError = error => ({
  type: "SET_ERROR",
  error
});

export const setLoading = loading => ({
  type: "SET_LOADING",
  loading
});

export const setPalette = palette => ({
  type: "SET_PALETTE",
  palette
});

export const setLock = colorID => ({
  type: "SET_LOCK",
  colorID
});

export const setModal = status => ({
  type: "SET_MODAL",
  status
});

export const setProjects = projects => ({
  type: "SET_PROJECTS",
  projects
});
