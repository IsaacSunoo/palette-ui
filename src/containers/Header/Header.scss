.Header {
  display: flex;
  height: 9%;
  border-bottom: #231f20 solid 3px;
  background: #21222D;
  color: #fff;
  justify-content: space-between;
  align-items: center;
  padding: 2em;

  h2 {
    margin: 0;
  }

  .btnContainer {

    button {
      width: 45px;
      outline: none;
      background-color: #F18805;
      font-family: $fontRobotoCond;
      color: #fff;
      border: none;
      font-size: 1.5em;
      line-height: 1.5;
      cursor: pointer;
      opacity: 1;
      transition: all .5s ease;
      box-shadow: 0px 0px 9px 2px #34a0e22e;

      &:hover {
        opacity: .6;
      }
    }

    button:first-child {
      border-top-left-radius: 35px;
      border-bottom-left-radius: 35px;
      padding-right: 0em;
      margin-right: .1em;
    }

    button:last-child {
      border-top-right-radius: 35px;
      border-bottom-right-radius: 35px;
      padding-left: 0em;
    }

  }


}
