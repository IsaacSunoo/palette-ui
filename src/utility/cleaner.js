export const cleanProjectsPalettes = (projects, palettes) => {
  const cleanedData = projects.map(project => {
    const { id, name } = project;
    let projectData =  { id, name, palettes: [] }
    palettes.forEach(palette => {
      if(palette.project_id === project.id){
        projectData.palettes.push(palette)
      }
    });
    return projectData;
  });
  return cleanedData;
}
