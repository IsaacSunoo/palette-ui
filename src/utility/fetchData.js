export const fetchData = async (url, body) => {
  try {
    const response = await fetch(url, body)
    const data = await response.json()
    if(!response.ok) { throw new Error(`'Fetch Call Cannot Be Made' ${data}`)}
    return data;
  } catch (error) {
    return error;
  }
}
