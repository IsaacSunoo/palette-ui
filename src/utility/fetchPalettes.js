export const fetchPalettes = async (url) => {
  try {
    const response = await fetch(url)
    if(!response.ok) { throw new Error('Fetch Call Cannot Be Made')}
    const data = await response.json()
    return data;
  } catch (error) {
    console.log(error);
    // this.props.setError('Fetch Call Cannont')
  }
}
