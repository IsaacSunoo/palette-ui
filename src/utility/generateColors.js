export const generateColors = (colorPalette) => {
  console.log(colorPalette);
  let newPalette = [];
  for(let i = 0; i < 5; i++) {
    const { color, locked } = colorPalette[i]
    if(colorPalette[i].locked) {
      newPalette.push({ color: color, locked: locked })
    } else {
      let randomColor = "#000000".replace( /0/g, function() { return (~~(Math.random()*16)).toString(16)});
      newPalette.push({ color: randomColor, locked: false })
    }
  }
  return newPalette;
}
