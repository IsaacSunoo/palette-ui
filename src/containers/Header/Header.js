import React, { Component } from 'react';
import { connect } from 'react-redux';

import * as actions from '../../actions/index';
import { generateColors } from '../../utility/generateColors';
import { Tooltip } from '@material-ui/core';

export class Header extends Component {
  constructor() {
    super()
    this.state = {
      color: ''
    }
  }

  handleGenerate = () => {
    const { currentPalette, setPalette } = this.props;
    const randomColors = generateColors(currentPalette)
    setPalette(randomColors);
  }

  saveButton = () => {
    const { modal, setModal } = this.props;
    setModal(modal)
  }

  render() {
    return (
      <div className='Header'>
        <div>
          <h2>Palette Picker</h2>
        </div>
        <div className='btnContainer'>
          <Tooltip title='Save Palette To Project'>
            <button onClick={this.saveButton}><i className='fas fa-save'></i></button>
          </Tooltip>
          <Tooltip title='Generate Colors!'>
            <button onClick={this.handleGenerate}><i className='fas fa-plus'></i></button>
          </Tooltip>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  currentPalette: state.currentPalette,
  modal: state.modalDisplay,
});

export const mapDispatchToProps = dispatch => ({
  setPalette: data => dispatch(actions.setPalette(data)),
  setModal: data => dispatch(actions.setModal(data)),
});

Header.propTypes = {

}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
