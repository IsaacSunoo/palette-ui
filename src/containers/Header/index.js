import React, { Component } from 'react';
import { connect } from 'react-redux';

import * as actions from '../../actions/index';
import { generateColors } from '../../utility/generateColors';

export class Header extends Component {

  handleGenerate = () => {
    const { currentPalette, setPalette } = this.props;
    const randomColors = generateColors(currentPalette)
    setPalette(randomColors);
  }

  render() {
    return (
      <div className="Header">
        <div>
          <h2>Palette Picker</h2>
        </div>
        <div>
          <button onClick={this.handleGenerate}>Generate</button>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  currentPalette: state.currentPalette
})

export const mapDispatchToProps = (dispatch) => ({
  setPalette: (data) => dispatch(actions.setPalette(data)),
})

Header.propTypes = {

}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
