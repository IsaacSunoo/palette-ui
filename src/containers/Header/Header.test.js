import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { generateColors } from '../../utility/generateColors';
jest.mock('../../utility/generateColors');
import * as actions from '../../actions';

const mockCurrentPalette = [
  { color: '#fffff', locked: false },
  { color: '#00000', locked: false },
  { color: '#231f20', locked: false },
  { color: '#e6e6e6', locked: false },
  { color: '#cccccc', locked: false }
];

describe('Header', () => {

  describe('Header Component', () => {
    let wrapper;
    let mockSetModal = jest.fn();
    let mockSetPalette = jest.fn();
    let mockModal = false;

    beforeEach(() => {
      wrapper = shallow(
        <Header currentPalette={mockCurrentPalette} setModal={mockSetModal} setPalette={mockSetPalette} modal={mockModal} />
      )
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should handleGenerate when clicked and call setPalette', async () => {
      generateColors.mockReturnValue(mockCurrentPalette);
      const button = wrapper.find('.generateBtn');
      button.simulate('click');
      actions.setPalette(mockCurrentPalette);
      expect(mockSetPalette).toHaveBeenCalledWith(mockCurrentPalette);
    });

    it('should setModal when saveButton is clicked', () => {
      const button = wrapper.find('.saveBtn');
      button.simulate('click');
      actions.setModal(mockModal);
      expect(mockSetModal).toHaveBeenCalledWith(mockModal);
    });

  });

  describe('mapStateToProps', () => {
    it('should return an object with currentPalettes', () => {
      const mockState = {
        currentPalette: ['#FFFFFF, #0000000'],
        modal: false
      };
      const expectedState = {
        currentPalette: ['#FFFFFF, #0000000']
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch for setPalette', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = actions.setPalette(mockCurrentPalette);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setPalette(mockCurrentPalette);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch for setModal', () => {
      const mockDispatch = jest.fn();
      const mockStatus = false;
      const actionToDispatch = actions.setModal(mockStatus);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setModal(mockStatus);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

  })

});
