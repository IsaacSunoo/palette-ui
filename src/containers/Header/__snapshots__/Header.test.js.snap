// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Component should match the snapshot 1`] = `
<div
  className="Header"
>
  <div>
    <h2>
      Palette Picker
    </h2>
  </div>
  <div
    className="btnContainer"
  >
    <WithStyles(Tooltip)
      title="Save Palette To Project"
    >
      <button
        className="saveBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-save"
        />
      </button>
    </WithStyles(Tooltip)>
    <WithStyles(Tooltip)
      title="Generate Colors!"
    >
      <button
        className="generateBtn"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus"
        />
      </button>
    </WithStyles(Tooltip)>
  </div>
</div>
`;
